pkg/node/manager/manager.go

```diff
@@ -127,6 +129,24 @@ type manager struct {
 
 	// healthScope reports on the current health status of the node manager module.
 	healthScope cell.Scope
+
+	// nodeNeighborQueue tracks node neighbor link updates.
+	nodeNeighborQueue queue[nodeQueueEntry]
+}
+
+type nodeQueueEntry struct {
+	node    *nodeTypes.Node
+	refresh bool
+}
+
+// Enqueue add a node to a controller managed queue which sets up the neighbor link.
+func (m *manager) Enqueue(n *nodeTypes.Node, refresh bool) {
+	if n == nil {
+		log.WithFields(logrus.Fields{
+			logfields.LogSubsys: "enqueue",
+		}).Warn("Skipping nodeNeighbor insert: No node given")
+	}
+	m.nodeNeighborQueue.push(&nodeQueueEntry{node: n, refresh: refresh})
```

Comment:
Is this function safe to be called concurrently? From what I understand - this can be called from different places, including the node handler code and the refresh neighbours controller?

Reply:
yes - the queue has mutex checks and the controller reconciles the state.